OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -287,7 +287,15 @@ void AddAsset(ScrollPanelWidget list, string filepath, IReadOnlyPackage package,
 				() => currentFilename == filepath && currentPackage == package,
 				() => { LoadAsset(package, filepath); });
 
-			item.Get<LabelWidget>("TITLE").GetText = () => filepath;
+			var labelWidget = item.Get<LabelWithTooltipWidget>("TITLE");
+			var label = WidgetUtils.TruncateLabelWithTooltip(labelWidget, filepath);
+
+			labelWidget.GetText = () => label;
```

Comment:
All this should go in the helper too!

Reply:
Got it. But then the helper had to return the whole widget. And to attach the new label in place of the old one I added a `ReplaceChild` method to `Widget`.

Reply:
I don't understand this. `LabelWidget` is a reference type, so there should be no need to return its value or replace it - it doesn't change!

Reply:
I'm sorry, learning C# as I go... I'll fix that later.

Reply:
Fixed.